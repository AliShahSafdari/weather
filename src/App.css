.App {
  text-align: center;
  background-color: #d44286;
  font-family: Gill Sans, Gill Sans MT, Calibri, Trebuchet MS, sans-serif;
}
