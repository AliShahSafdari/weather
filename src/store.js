import { configureStore } from '@reduxjs/toolkit';
import weathersReducer from './redux/weather/weathersReducer';

const store = configureStore({
  reducer: {
    weathers: weathersReducer,
  },
});

export default store;
