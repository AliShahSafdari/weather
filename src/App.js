import { Route, Routes } from 'react-router-dom';
import ShowAll from './components/HomePage';
import CityDetails from './components/DetailPage';
import PageNotFound from './components/NotFound';
import './App.css';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<ShowAll />} />
        <Route path="/details/:cityId" element={<CityDetails />} />
        <Route path="*" element={<PageNotFound />} />
      </Routes>
    </div>
  );
}

export default App;
