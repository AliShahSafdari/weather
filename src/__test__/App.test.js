import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import store from '../store';

describe('Component render testing', () => {
  test('App page renders correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
