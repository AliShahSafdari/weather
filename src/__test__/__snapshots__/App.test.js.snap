// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render testing App page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <header
        class="header"
      >
        <h1
          class="header-title"
        >
          WeatherHub: Explore Weather in Major Cities 
        </h1>
        <div
          class="header-content"
        >
          <div
            class="header-filter"
          >
            <span>
              Filter
            </span>
            <select>
              <option
                value="All"
              >
                All
              </option>
              <option
                value="Hot"
              >
                Hot
              </option>
              <option
                value="Temperate"
              >
                Temperate
              </option>
              <option
                value="Cold"
              >
                Cold
              </option>
            </select>
          </div>
          <h2>
            Most views
          </h2>
          <div
            class="icons"
          >
            <img
              alt="microphone"
              class="icons-img"
              src="blue-radio-microphone.png"
            />
            <img
              alt="setting"
              class="icons-img"
              src="settings.png"
            />
          </div>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
